src/SearchBar.jsx

```diff
@@ -1,66 +1,168 @@
-import React from 'react';
+import PropTypes from 'prop-types'
+import React from 'react'
 import { connect } from 'react-redux'
 import search from './actions/search'
-import './SearchBar.less';
+import Typeahead, { MODE_OPEN } from './Typeahead'
+import HighlightingOption from './Typeahead/HighlightingOption'
+import './SearchBar.less'
+
+const searchFields = {
+  'all': 'All Data',
+  'company': 'Company Name',
+  'complaint_what_happened': 'Narratives'
+}
 
 export class SearchBar extends React.Component {
   constructor(props) {
-    super(props);
+    super(props)
     this.state = {
-      inputValue: props.searchText
-    };
+      inputValue: props.searchText,
+      searchField: props.searchField
+    }
 
     // This binding is necessary to make `this` work in the callback
     // https://facebook.github.io/react/docs/handling-events.html
-    this._handleSubmit = this._handleSubmit.bind(this);
-    this._updateInputValue = this._updateInputValue.bind(this);
+    this._handleSubmit = this._handleSubmit.bind(this)
+    this._onInputChange = this._onInputChange.bind(this)
+    this._onSelectSearchField = this._onSelectSearchField.bind(this)
+    this._onTypeaheadSelected = this._onTypeaheadSelected.bind(this)
   }
 
   componentWillReceiveProps(nextProps) {
     this.setState({
-      inputValue: nextProps.searchText
-    });
-  }
-
-  _handleSubmit(event) {
-    event.preventDefault();
-    this.props.onSearch(this.state.inputValue);
+      inputValue: nextProps.searchText,
+      searchField: nextProps.searchField
+    })
   }
 
-  _updateInputValue(evt) {
-    this.setState({
-      inputValue: evt.target.value
-    });
+  // This prevents a duplicate update that seems to be triggered on page load
+  shouldComponentUpdate(nextProps, nextState) {
+    return JSON.stringify(this.state) !== JSON.stringify(nextState)
   }
 
   render() {
     return (
         <nav className="search-bar">
           <form action="" onSubmit={this._handleSubmit}>
-            <label className="a-label a-label__heading" htmlFor="searchText">Search Within</label>
-            <input type="button" className="btn" value="Types" />
-            <input type="text"
-                   value={this.state.inputValue}
-                   onChange={this._updateInputValue}
-                   placeholder="Enter your search term(s)"
-                   id="searchText" />
-            <input type="submit" className="btn primary" value="Search" />
+            <h5>Search Within</h5>
+            <div className="layout-row">
+              <div className="cf-select">
```

Comment:
The `ask-cfpb` [page](https://www.consumerfinance.gov/ask-cfpb/category-student-loans/) is using a type-a-head as well. 

<img width="1007" alt="screen shot 2017-08-02 at 4 50 33 am" src="https://user-images.githubusercontent.com/1696212/28865546-43f7fe86-773e-11e7-9991-b495bda6296f.png">


It probably makes sense to piggy back on that markup that markup